apollo-router/src/plugins/rhai/mod.rs

```diff
@@ -1093,79 +1110,72 @@ impl ServiceStep {
     }
 }
 
-struct ErrorDetails {
-    status: StatusCode,
-    message: String,
-    position: Option<Position>,
-    // Add support for extension_code ?
+#[derive(Deserialize, Debug)]
+struct Position {
+    line: Option<usize>,
+    pos: Option<usize>,
 }
 
-impl fmt::Display for ErrorDetails {
+impl fmt::Display for Position {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self.position {
-            Some(pos) => {
-                write!(f, "{}: {}({})", self.status, self.message, pos)
-            }
-            None => {
-                write!(f, "{}: {}", self.status, self.message)
-            }
+        if self.line.is_none() || self.pos.is_none() {
+            write!(f, "none")
+        } else {
+            write!(
+                f,
+                "line {}, position {}",
+                self.line.expect("checked above;qed"),
+                self.pos.expect("checked above;qed")
+            )
+        }
+    }
+}
+
+impl From<&rhai::Position> for Position {
+    fn from(value: &rhai::Position) -> Self {
+        Self {
+            line: value.line(),
+            pos: value.position(),
         }
     }
 }
 
+#[derive(Deserialize, Debug)]
+struct ErrorDetails {
+    #[serde(
```

Comment:
do we really need to bring in http_serde just to serialize the status code?

Reply:
I don't have any other way in my mind, except implementing it manually :/